Rubberduck.VBEEditor/Extensions/DictionaryExtensions.cs

```diff
@@ -24,16 +24,27 @@ public static class DictionaryExtensions
             return source.SelectMany(item => item.Value);
         }
 
+        public static IEnumerable<TValue> AllValues<TKey1, TKey2, TValue>(
+            this IDictionary<TKey1, IDictionary<TKey2, List<TValue>>> source)
+        {
+            return source.SelectMany(item => item.Value.AllValues());
+        }
+
         public static ConcurrentDictionary<TKey, ConcurrentBag<TValue>> ToConcurrentDictionary<TKey, TValue>(this IEnumerable<IGrouping<TKey, TValue>> source)
         {
             return new ConcurrentDictionary<TKey, ConcurrentBag<TValue>>(source.Select(x => new KeyValuePair<TKey, ConcurrentBag<TValue>>(x.Key, new ConcurrentBag<TValue>(x))));
         }
 
-        public static Dictionary<TKey, List<TValue>> ToDictionary<TKey, TValue>(this IEnumerable<IGrouping<TKey, TValue>> source)
+        public static IDictionary<TKey, List<TValue>> ToDictionary<TKey, TValue>(this IEnumerable<IGrouping<TKey, TValue>> source)
         {
             return source.ToDictionary(group => group.Key, group => group.ToList());
         }
 
+        public static IDictionary<TKey1, IDictionary<TKey2, List<TValue>>> ToDictionary<TKey1, TKey2, TValue>(this IEnumerable<IGrouping<TKey1, IGrouping<TKey2, TValue>>> source)
+        {
+            return source.ToDictionary(group => group.Key, group => group.ToDictionary());
+        }
```

Comment:
Should these one-liner `return` statement methods be expression-bodied?

Reply:
It currently follows the style of the other methods in the class.

Reply:
Should I convert all functions in the class to be expression bodied?

Reply:
I'm making such one-liners expression-bodied when I see opportunities, but I wouldn't say it's technical debt that really really needs to be paid - your call =)

Reply:
I think I will not change them to expression-bodied functions since it does not help readability the lines being as long as they are.